dask/highlevelgraph.py

```diff
@@ -447,6 +469,66 @@ def cull(self, keys: Set):
 
         return HighLevelGraph(ret_layers, ret_dependencies, ret_key_deps)
 
+    def map_basic_layers(
+        self, func: Callable[[BasicLayer], Mapping]
+    ) -> "HighLevelGraph":
+        """Map `func` on each basic layer and returns a new high level graph.
+
+        `func` should take a BasicLayer as input and return a new Mapping as output
+        and **cannot** change the dependencies between Layers.
+
+        If `func` returns a non-BasicLayer type, it will be wrapped in a `BasicLayer`
+        object automatically.
+
+        Parameters
+        ----------
+        func : callable
+            The function to call on each BasicLayer
+
+        Returns
+        -------
+        hlg : HighLevelGraph
+            A high level graph containing the transformed BasicLayers and the other
+            Layers untouched
+        """
+        layers = {}
+        for k, v in self.layers.items():
+            if isinstance(v, BasicLayer):
+                layer = func(v)
+                if not isinstance(layer, BasicLayer):
+                    layer = BasicLayer(layer)
```

Comment:
Could you elaborate on why we always need to return a `BasicLayer`? We may want the flexibility to have `func` return `Layer`s of different types. Though we can certainly hold off on that until the need arises : )

Reply:
No, you are absolutely right no need to restrict this to `BasicLayer` :)
Fixed